src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.DecisionDagRewriter.cs

```diff
@@ -904,7 +904,7 @@ private void LowerWhenClause(BoundWhenDecisionDagNode whenClause)
 
                     // We hide the jump back into the decision dag, as it is not logically part of the when clause
                     BoundStatement jump = _factory.Goto(GetDagNodeLabel(whenFalse));
-                    sectionBuilder.Add(GenerateSequencePoints ? _factory.HiddenSequencePoint(jump) : jump);
+                    sectionBuilder.Add(GenerateInstrumentation ? _factory.HiddenSequencePoint(jump) : jump);
```

Comment:
>GenerateInstrumentation [](start = 39, length = 23)

It is not clear why generation of hidden sequence points is conditioned on instrumentation

Reply:
Because we produce sequence points whenever we use the instrumentation API and only then.  Hidden sequence points are needed here when we are producing sequence points.  (When not producing a pdb, the hidden sequence points are dropped on the floor later along with the sequence points).

---
In reply to: [412352452](https://github.com/dotnet/roslyn/pull/43430#discussion_r412352452) [](ancestors = 412352452)